lib/resources/user-collection.js

```diff
@@ -166,6 +138,78 @@ UserCollection.prototype.handle = function (ctx) {
   }
 };
 
+function handleLogin(uc, ctx) {
+  var path = uc.path
+    , credentials = ctx.req.body || {};
+  
+  debug('trying to login as %s', credentials.username);
+  
+  uc.store.first({ username: credentials.username }, function (err, user) {
+    if (err) return ctx.done(err);
+    // keep a clone of the user so we can compare it later to see if any changes were made in the login event
+    var userClone = user ? _.clone(user) : null
+      , domain = { 'me': userClone, 'data': userClone, 'success': false };
+    
+    // checks if the user was changed in the login event and saves it if it was
+    function saveUser(fn) {
```

Comment:
shall we call this as ```checkAndSaveUser```?

And move loginDone, loginFail and saveUser to another function 
```UserCollection.prototype.afterLoginEvent = function(){}```

Reason:

In my case, I call 
```
uc.store.first({$or:[{username:credentials.username}, {tel:credentials.username}, {email:credentials.username}]}, 
```
in order to login by email OR tel OR username. 
And also handle forget password and robot access in here. 

I would like to do that by overriding ```handleLogin```, and then, I can call  the Login event and afterLoginEvent function.